deploy/helm/kubecf/assets/operations/instance_groups/credhub.yaml

```diff
@@ -111,8 +111,11 @@
   type: remove
 - path: /instance_groups/name=api/jobs/name=cloud_controller_ng/properties/credhub_api?
   type: remove
-- path: /instance_groups/name=diego-cell/jobs/name=cflinuxfs3-rootfs-setup/properties/cflinuxfs3-rootfs/trusted_certs/1
-  type: remove
+# The cflinuxfs3-rootfs/properties are being replaced by operations/stacks.yaml and don't include credhub CA unless enabled
+#- path: /instance_groups/name=diego-cell/jobs/name=cflinuxfs3-rootfs-setup/properties/cflinuxfs3-rootfs/trusted_certs/1
+#  type: remove
+# XXX Is there a safer way to do this than to delete an array element by position?
+# XXX This could break silently during any cf-deployment upgrade.
```

Comment:
I don't think that's possible; `k=v` matches in go-patch needs array of maps.  Otherwise it's just array by index and map by key.

But given that we clobber all of it I am not sure why that matters?

Reply:
> But given that we clobber all of it I am not sure why that matters?

Unlike the `cflinuxfs3-rootfs-setup` job we are not clobbering the whole thing for the `rep` job, we are removing the 2nd element of the array:

```
- path: /instance_groups/name=diego-cell/jobs/name=rep/properties/containers/trusted_ca_certificates/1
  type: remove
```

which is editing this:

```
  - name: rep
    release: diego
    properties:
      containers:
        trusted_ca_certificates:
        - ((diego_instance_identity_ca.ca))
        - ((credhub_tls.ca))
        - ((uaa_ssl.ca))
```

So if certificates are re-orderd, or added, then the index `/1` will become wrong.

The `XXX` comments apply to the lines below, and not the commented lines above.

One thing we _could_ do is look inside `cf-deployment.yml` and verify that this array element still contains the string `((credhub_tls.ca))` and `{{ fail }}` if it doesn't.

Reply:
Ah, given that the comment immediately followed the previous, commented out operation (with no blank lines) I thought it applied to that.  It may be useful to delete the unused operation, to make it obvious that it applies to `rep`?

Reply:
Ok, will update to make it clearer. I did comment and not remove the other part because I wanted to leave a tombstone, to avoid somebody adding this code back by accident in the future. But I'll just leave a comment instead and will remove the code/ops.

Reply:
Actually just inserting a new line between your tombstone and the `XXX` referring to the following lines should be enough.

Reply:
Too late, already updated in 30cf1ea